atlas-db-controller/controller.go

```diff
@@ -639,8 +647,20 @@ func (c *Controller) syncDatabaseSecret(key string, db *atlas.Database, dbserver
 			}
 			if errors.IsNotFound(errs) {
 				glog.V(4).Info("Database secrets not found. Creating...")
-				dsn := server.ActivePlugin(dbserver).Dsn(user.Name, passwd, db, dbserver)
-				secret, errs = c.kubeclientset.CoreV1().Secrets(db.Namespace).Create(
+				if dbServer != nil {
+					dsn = dbPlugin.Dsn(user.Name, passwd, db, dbServer)
+				} else {
+					customDbServer := &atlas.DatabaseServer{}
+					spec := atlas.DatabaseServerSpec{}
+					customDbServer.Spec = spec
+					splitDSN := strings.Split(strings.Split(dsn, "@")[1], "/")[0]
```

Comment:
better move this to a utility method which extracts details from dsn string, would be useful